src/common/ownsql.cpp

```diff
@@ -188,7 +188,8 @@ QString SqlDatabase::error() const
 void SqlDatabase::close()
 {
     if (_db) {
-        foreach (auto q, _queries) {
+        while (!_queries.empty()) {
+            auto q = *_queries.begin();
```

Comment:
Slightly stylistic but with a Qt container like this I'd use `.first()` instead of going through iterator (which detached) + dereferencing

Reply:
`QSet<T>` (which `_queries` is) has no `.first()` member function. As I understand the Qt docs, the original version would also detach, but only once. (My version detaches every iteration, and then throws away the detached copy, if I get this correctly.)

Something like this is probably better:
```c++
        const auto queries = _queries;
        for (auto & q : queries) {
            q->finish();     // This will detach queries from _queries, but only the first time.
        }
```

Reply:
Oh right that's a `QSet<T>`, my bad. :-)

Since my comment was mostly stylistic I would say: ignore me on that one. Also, I realize I was wrong about the detach part of my comment since we operate on _queries directly. It's just that the iterator are invalidated when calling `finish()` but we don't keep them around we're likely fine.

Sorry for the noise.

Reply:
Ack, I'll leave it as is then,

The real question is if there's a reason this is a set and not a vector, but that's for another issue or PR, methinks.

Reply:
> Ack, I'll leave it as is then,
> 
> The real question is if there's a reason this is a set and not a vector, but that's for another issue or PR, methinks.

Yeah thought the same... a `QVector` might be a better choice, I don't see much advantage in the hashing there.